sunpy/map/tests/test_map_factory.py

```diff
@@ -98,6 +102,11 @@ def test_patterns(self):
         assert isinstance(maps, list)
         assert ([isinstance(amap, sunpy.map.GenericMap) for amap in maps])
 
+        # File-like object
+        with open(a_fname, "rb") as fd:
```

Comment:
What happens if you pass random binary data to map?

Reply:
it returns a `ValueError`

Reply:
Is there a test for this in this file?

Reply:
Not currently, should I add one?

Reply:
If you could.